db/db.go

```diff
@@ -68,19 +64,58 @@ type DB struct {
 
 // NewDB creates a new DB, which will *own* ds and dispatcher for internal use.
 // Saying it differently, ds and dispatcher shouldn't be used externally.
-func NewDB(ts service.Service, opts ...Option) (*DB, error) {
+func NewDB(ctx context.Context, ts s.Service, id thread.ID, opts ...Option) (*DB, error) {
 	config := &Config{}
 	for _, opt := range opts {
 		if err := opt(config); err != nil {
 			return nil, err
 		}
 	}
-	return newDB(ts, config)
+
+	ti, err := ts.GetThread(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if len(ti.Logs) == 0 {
+		ti, err = ts.CreateThread(ctx, id, s.FollowKey(sym.New()), s.ReadKey(sym.New()))
+		if err != nil {
+			return nil, err
+		}
+	}
+	return newDB(ts, id, config)
+}
+
+// NewDBFromAddr creates a new DB from a thread hosted by another peer at address,
+// which will *own* ds and dispatcher for internal use.
+// Saying it differently, ds and dispatcher shouldn't be used externally.
+func NewDBFromAddr(ctx context.Context, ts s.Service, addr ma.Multiaddr, followKey, readKey *sym.Key, opts ...Option) (*DB, error) {
```

Comment:
Related to the question I asked in the client layer (about no thread id being passed in here), I guess it's because the thread id was already created on another peer?

Reply:
Yep. The thread ID is part of the address, which was created by a different peer.